### ISSUE TITLE
About dataset

### ISSUE THREAD

**Issue by ProudZhangXiaolong** [szzexpoi/POEM#5]
How to get **novel_vqa_val_known_question.json** and **novel_vqa_val_known_annotation.json**?

**Comment by ProudZhangXiaolong** [comment]
Thank you very much !!

**Comment by szzexpoi** [comment]
You can find the data here, as listed in the instruction of the data folder:

https://drive.google.com/file/d/1dTg5Dn1BmCiwY_gXOG06lo60LOZJrw29/view?usp=drivesdk

**Comment by ProudZhangXiaolong** [comment]
Thank you very much !!

**Comment by ProudZhangXiaolong** [comment]
![6](https://github.com/szzexpoi/POEM/assets/62319416/e955d6ff-27e9-4b7d-9895-cf0f3d8b3fe0)
Excuse me, where does the**. npy file** come from? Is it processed by **trainval_feature.h5**

**Comment by ProudZhangXiaolong** [comment]
![6](https://github.com/szzexpoi/POEM/assets/62319416/5d2140cb-a710-476a-a49e-acb8476ae40c)
Excuse me, where does the**. npy file** come from? Is it processed by **trainval_feature.h5**

**Comment by ProudZhangXiaolong** [comment]
![6](https://github.com/szzexpoi/POEM/assets/62319416/42a9a5d7-6ac1-495a-9451-440215e84cac)
Excuse me, where does the **npy file** come from? Is it processed by **trainval_feature.h5**?

**Comment by szzexpoi** [comment]
Yes, please refer to the last several lines of the instruction in the data folder.

**Comment by ProudZhangXiaolong** [comment]
Thank you!

**Comment by ProudZhangXiaolong** [comment]
I'm really sorry to bother you again. I want to retrain **proto_learning**. When I was running [extract-vqa. py](https://github.com/szzexpoi/POEM/blob/main/data/extract_vqa.py), _np.frombuffer (base64.decodestring)..._ encountered an error and decoding problem was difficult to solve. Could you share the complete weight file of [**Prototype_Module**](proto_learning/module.py) to predict labels? (on all datasets).
Thank you!
My email: zjinhao2022@126.com
![Snipaste_2024-07-25_23-53-31](https://github.com/user-attachments/assets/89cf385e-d995-4e63-9f7c-c7553380c1cb)

**Comment by szzexpoi** [comment]
You probably need to run that code with python2 (maybe). The complete weights for different datasets can be found in the link under the proto_learning folder:
https://drive.google.com/file/d/1alCvI8tub2yv0lJI2shjTlC8fA_TEZLe/view

**Comment by ProudZhangXiaolong** [comment]
I'm very sorry to bother you again
May I ask if it's possible to share the **model_best.pt** of the [**README.md**](https://github.com/szzexpoi/POEM/blob/main/proto_learning/README.md) under the **proto_learning** file
>>> import torch
>>> proto = torch.load('./ckpt/model_best.pt')['prototype.weight']
>>> torch.save(proto,'prototype.pt')

![Snipaste_2024-07-26_10-02-00](https://github.com/user-attachments/assets/613afba5-2542-4932-8e6b-2d317b7de4f6)
Thank you very much !

**Comment by szzexpoi** [comment]
The "model_best.pt" in the instruction is just an example, you should replace it with the weights for different datasets provided in Google drive. Sorry for the confusion.

**Comment by ProudZhangXiaolong** [comment]
I'm sorry to bother you again, perhaps you misunderstood my meaning. 
The shape of the parameter of **prototype_vqa.pt** you provided is [1000,2048], which corresponds to the parameter of the **self.prototype** in the [Prototype Module class](proto_learning/module.py). 
However, I need the complete parameters of the [Prototype Module class](proto_learning/module.py) including **self.prototype**, **self.proto2concept**, and **self.attention_layer**.
Thank you very much for sharing. I apologize for bothering you multiple times！

**Comment by szzexpoi** [comment]
I see. In that case, I may not be able to help. Those weights are not used in the final models; thus, I did not save them in Google Drive. I graduated a year ago and can no longer access the previous lab servers (those weights are probably wiped out as well).

**Comment by ProudZhangXiaolong** [comment]
Anyway, thanks a million!
